csrc/host_ir/executor.cpp

```diff
@@ -722,6 +725,23 @@ void HostIrEvaluator::handle(LoadStoreOp* load_store_op) {
   }
 }
 
+#ifdef NVFUSER_HOST_IR_JIT
+void HostIrEvaluator::handle(kir::Allocate* allocate) {
+  FUSER_PERF_SCOPE("HostIrEvaluator::handle(kir::Allocate)");
+  NVF_ERROR(
+      allocate->buffer()->isA<TensorView>(),
+      "Allocation must be on a TensorView but got ",
+      allocate->buffer());
+  TensorView* tv = allocate->buffer()->as<TensorView>();
+  if (expr_evaluator_.isKnown(tv)) {
+    return;
+  }
+  auto shape_info = inferTensorShapes(tv, expr_evaluator_);
```

Comment:
nitpick, the above code is shared between the macro guard.
You can put the macro right before this line to avoid some code duplication.